File: solved/133A.rb
p = gets.chomp
a = p.scan(/./)
f = 0
a.each do |c|
  if c == "H" || c == "Q" || c == "9"
    f = 1
    break
  end
end

if f == 1
  puts "YES"
else
  puts "NO"
end

File: solved/133A.cc
#include <iostream>
#include <string>

using namespace std;

int main() {
  string s;
  cin >> s;
  int sl = s.length();
  for(int i=0; i<sl; ++i)
    if(s[i] == 'H' || s[i] == 'Q' || s[i] == '9') {
      cout << "YES";
      return 0;
      }

  cout << "NO";
  return 0;
  }
